lib/IDE/Refactoring.cpp

```diff
@@ -6847,15 +7038,18 @@ class AsyncConverter : private SourceEntityWalker {
     auto InlinePatterns =
         getInlinePatternsToPrint(Blocks.SuccessBlock, SuccessParams, Callback);
 
-    prepareNames(Blocks.SuccessBlock, SuccessParams, InlinePatterns);
-    preparePlaceholdersAndUnwraps(HandlerDesc, SuccessParams, ErrParam,
-                                  /*Success=*/true);
+    {
+      llvm::SaveAndRestore<bool> InSuccesBranch(ConvertingSuccessBlocks, true);
```

Comment:
What do you think about moving this into `convertNodes`, and having `convertNodes` take a bool param for success or failure? That way if we add more uses of `convertNodes` in the future we won't forget to update this state.

Reply:
Good idea, because I only now realized that we weren’t setting `ConvertingSuccessBlocks` to `false` when converting an error block nested in a success block. So the suggestion already caught a potential bug.